web/src/sidebar_ui.js

```diff
@@ -32,31 +37,68 @@ export function any_sidebar_expanded_as_overlay() {
 }
 
 export function initialize() {
-    $("body").on("click", ".login_button", (e) => {
-        e.preventDefault();
-        e.stopPropagation();
-        window.location.href = hash_util.build_login_link();
-    });
-
-    $("#userlist-toggle-button").on("click", (e) => {
-        e.preventDefault();
-        e.stopPropagation();
-
-        const sidebarHidden = !$(".app-main .column-right").hasClass("expanded");
-        popovers.hide_all();
-        if (sidebarHidden) {
-            show_userlist_sidebar();
-        }
-    });
-
-    $("#streamlist-toggle-button").on("click", (e) => {
-        e.preventDefault();
-        e.stopPropagation();
-
-        const sidebarHidden = !$(".app-main .column-left").hasClass("expanded");
-        popovers.hide_all();
-        if (sidebarHidden) {
-            show_streamlist_sidebar();
-        }
-    });
+    document.addEventListener(
+        "click",
+        (e) => {
+            // This check is just an optimization since this is called on every click.
+            if (window.innerWidth > media_breakpoints_num.xl) {
+                // Sidebars are always visible beyond xl breakpoint.
+                return;
+            }
+
+            const $left_column = $(".app-main .column-left");
+            const $right_column = $(".app-main .column-right");
+            const $elt = $(e.target);
+            const toggle_left_sidebar = Boolean($elt.closest("#streamlist-toggle-button").length);
+            const toggle_right_sidebar = Boolean($elt.closest("#userlist-toggle-button").length);
+            const left_sidebar_visible = $left_column.hasClass("expanded");
+            const right_sidebar_visible = $right_column.hasClass("expanded");
+
+            function hide_left_sidebar() {
+                if (left_sidebar_visible) {
+                    $left_column.removeClass("expanded");
+                }
+            }
+
+            function hide_right_sidebar() {
+                if (right_sidebar_visible) {
+                    $right_column.removeClass("expanded");
+                }
+            }
```

Comment:
Can we use the functions like `show_userlist_sidebar` rather than directly adding/removing classes? It's cleaner and will be important if we add the variables suggested here: https://github.com/zulip/zulip/pull/26908/commits/65630fd5e0a806740353ba1b5d5ca561b925c0e1#r1341715178